### ISSUE TITLE
it seems you are training and validating on the same dataset?

### ISSUE THREAD

**Issue by sunwayforever** [ChunML/ssd-tf2#26]
although the ids is split into `train` and `val`,  you are always sampling  from the whole ids

**Comment by AkshayPS12** [comment]
Hi @sunwayforever if you see in `voc_data.py` in the `generate` function , the indices of `train` and `val` data are separated and they are being used for "sampling".

**Comment by showyamessage** [comment]
Yes, they are separated. But get_annotation and get_image functions get their filenames from self.ids. I think, transfering the filename (calculated in generate anyway) directly to get_annotation and get_image should solve this issue.
